### PR TITLE
[DPE-4456] Fix scale up with S3 and TLS relations

### PR DESCRIPTION
**Pull request by marceloneppel**
## Issue
We cannot scale up if S3 and TLS relations are established. This operation leads to an error like the one below if, in the new unit, the S3 parameters relation changed event fires before the TLS files are available in the unit (and Patroni + PostgreSQL haven't started yet).
```sh
requests.exceptions.SSLError: HTTPSConnectionPool(host='10.142.152.2', port=8008): Max retries exceeded with url: /cluster (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1007)')))
```

## Solution
Defer the S3 parameters relation changed event until that unit can do any operation related to S3 (or even connect to the other unit to check the cluster topology and verify it's not the primary unit so that it can abort any kind of stanza initialisation).

Port of https://github.com/canonical/postgresql-operator/pull/480.

### CODE REVIEW

File: src/backups.py [hunk 1]
```diff
@@ -455,6 +471,11 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
             logger.debug("Cannot set pgBackRest configurations, missing configurations.")
             return
 
+        if not self._can_initialise_stanza:
+            logger.debug("Cannot initialise stanza yet.")
```
Comment (lucasgameiroborges) [inline src/backups.py:475]: Should this be DEBUG level only? Kinda feel a little more important than this to me, but up to you!

**Review by lucasgameiroborges — APPROVED** [review]
LGTM! Left only one question.

**Review by taurus-forever — APPROVED** [review]
LGTM, but 30 mins timeout scary me a lot...

File: src/backups.py (hunk 1 above)
Comment (marceloneppel) [reply src/backups.py:475]: I think DEBUG is okay for this case. Like the other messages we already have for similar situations, which don't provide immediate value for the user but only when troubleshooting, I think it's okay as it is.

**Pull request merged**